Tests.ReactView/Sandbox.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Threading.Tasks;
+using Avalonia.Controls;
+
+namespace Tests.ReactView {
+
+    internal class Sandbox : IDisposable {
+
+        private TestReactView View { get; }
+
+        private Sandbox(Window window, string propertyValue) : this(propertyValue) {
+            AttachTo(window);
+        }
+
+        public Sandbox(string propertyValue) {
+            View = new TestReactView {
+                PropertyValue = propertyValue
+            };
+        }
+
+        public static async Task<Sandbox> InitializeAsync(Window window, string propertyValue) {
+            var sandbox = new Sandbox(window, propertyValue);
+            await sandbox.Ready();
+
+            return sandbox;
+        }
+
+        public void AttachTo(Window window) {
+            window.Content = View;
+        }
+
+        public string GetFirstRenderHtml() {
+            return View.EvaluateMethod<string>("getFirstRenderHtml");
+        }
+
+        public string GetHtml() {
+            return View.EvaluateMethod<string>("getHtml");
+        }
+
+        public string GetPropertyValue() {
+            return View.EvaluateMethod<string>("getPropertyValue");
+        }
+
+        public string PropertyValue { 
+            get => View.PropertyValue; 
+            set => View.PropertyValue = value; 
+        }
+
+        public Task Ready() {
+            var taskCompletionSource = new TaskCompletionSource<bool>();
+            void OnReady() {
+                try {
+                    taskCompletionSource.SetResult(true);
+                } finally {
+                    View.Ready -= OnReady;
+                }
+            }
+            View.Ready += OnReady;
+            return taskCompletionSource.Task;
```

Comment:
try finally should be here not inside the onready

Reply:
Then we have to await here instead

Reply:
Otherwise the finally block will run before it's supposed to

Reply:
no problem with that